src/content/app/tools/blast/views/blast-submission-results/BlastSubmissionResults.tsx

```diff
@@ -167,35 +182,275 @@ type SingleBlastJobResultProps = {
+          queryStart: hitHsp.hsp_query_from,
+          queryEnd: hitHsp.hsp_query_to,
+          hitStart: hitHsp.hsp_hit_from,
+          hitEnd: hitHsp.hsp_hit_to
+        };
+
+        counter++;
+      });
+    });
+
+    setTableData(allTableData);
+    setSequenceAlignmentData(newSequenceAlignmentData);
+  }, [data]);
+
+  const [expandedContent, setExpandedContent] = useState<{
+    [rowId: string]: ReactNode;
+  }>({});
+
+  const onExpanded = (isExpanded: boolean, rowId: string) => {
+    if (sequenceAlignmentData && sequenceAlignmentData[rowId]) {
+      const aligment = (
+        <div className={styles.sequenceAlignment}>
+          <BlastSequenceAlignment
+            alignment={sequenceAlignmentData[rowId]}
+            blastDatabase={blastDatabase}
+          />
+        </div>
+      );
+      setExpandedContent({
+        ...expandedContent,
+        [rowId]: isExpanded ? aligment : undefined
+      });
+    }
+  };
+
+  hitsTableColumns[3].renderer = (params: TableCellRendererParams) => {
+    return (
+      <ShowHideColumn
+        onExpanded={onExpanded}
+        rowId={params.rowId}
+        isExpanded={!!expandedContent[params.rowId]}
+      />
+    );
+  };
+
   if (!data) {
     return null;
   }
 
   const alignmentsCount = countAlignments(data.result);
 
   return (
-    <div className={styles.resultsSummaryRow}>
-      <div className={styles.hitLabel}>
-        <span>{alignmentsCount} </span>
-        <span className={styles.label}>
-          {`${pluralise('hit', alignmentsCount)}`}
-        </span>
-      </div>
-      <div className={styles.summaryPlot}>
-        <BlastHitsDiagram job={data.result} width={diagramWidth} />
-      </div>
-      <div className={styles.speciesInfo}>
-        {speciesInfo.common_name && <span>{speciesInfo.common_name}</span>}
-        <span>{speciesInfo.scientific_name}</span>
-        <span>{speciesInfo.assembly_name}</span>
+    <>
+      <div className={styles.resultsSummaryRow}>
+        <div className={styles.hitLabel}>
+          <span>{alignmentsCount} </span>
+          <span className={styles.label}>
+            {`${pluralise('hit', alignmentsCount)}`}
+          </span>
+        </div>
+        <div className={styles.summaryPlot}>
+          <BlastHitsDiagram job={data.result} width={diagramWidth} />
+        </div>
+        <div className={styles.speciesInfo}>
+          {speciesInfo.common_name && <span>{speciesInfo.common_name}</span>}
+          <span>{speciesInfo.scientific_name}</span>
+          <span>{speciesInfo.assembly_name}</span>
+          <div className={styles.showHideWrapper}>
+            <ShowHide
+              isExpanded={isExpanded}
+              onClick={() => setExpanded(!isExpanded)}
+            ></ShowHide>
+          </div>
+        </div>
+
+        {isExpanded && (
+          <div className={styles.tableWrapper}>
+            <DataTable
+              columns={hitsTableColumns}
+              data={tableData}
+              theme={'dark'}
```

Comment:
I naturally tent to use`{}` while writing the code. Is there any reason why we shouldn't use `{}`?

Reply:
`{}` is the JSX syntax for evaluating expressions; but strings do not need evaluation. In this regard, a JSX element behaves like a regular html element, whose attributes are strings. When you write `<img src="foo.jpg"/>`, it is valid JSX; and it's unlikely you would have written it `<img src={"foo.jpg"} />`. Same with any React component that can take string attributes, such as `<Thing foo="bar" />`.

Whether to write `<Thing foo="bar" />` or `<Thing foo={"bar"} />` is a matter of style. Obviously, both will work. To me, `<Thing foo="bar" />` is cleaner, because it does not introduce the unnecessary curly bracket noise. It's almost like writing `return x > y` instead of `if (x > y) { return true; } else { return false; }`.